mesonbuild/msubprojects.py

```diff
@@ -381,12 +414,14 @@ def download(self):
             return False
         return True
 
-    def foreach(self):
+    def foreach(self) -> bool:
+        options = T.cast(ForeachArguments, self.options)
```

Comment:
```suggestion
        options = T.cast('ForeachArguments', self.options)
```

This is only defined during type checking, so quoting it will ensure it is doing deferred evaluation.